Assets/HoloToolkit/Input/Scripts/InputSources/InputManagerExtensions.cs

```diff
@@ -0,0 +1,314 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    public static partial class InputManagerExtensions
```

Comment:
The managers in HoloToolit that are partial shouldn't be partial, IMO. It encourages bad design, and in most cases it isn't that helpful anyway. I'd rather we design our classes properly with proper support for inheritance and such. It was a conscious decision that the new version of GazeManager isn't partial.

In this specific case, if someone wanted to add more extensions to the InputManager, nothing prevents that person from creating his own extension class: there is no need to make InputManagerExtensions partial.

As for using extensions versus putting the methods into the class itself, I don't feel strongly about it, although I tend to avoid using extensions when in the same DLL / module. I think extension methods are great when you want to extend a class you can't modify: for example, if you have an external DLL providing class Foo and you want to add a method to it. In the case of InputManager, the implementation that is in InputManagerExtensions is actually part of the same "plugin" (HoloToolkit-Input), so we probably should add them directly into InputManager. 

I agree that should a developer want to add other methods without contributing back to HoloToolkit, he'd create an extension, but I don't think we should do that in our own code for core input events.

I'd remove the "partial" from the InputManagerExtensions. I'll be happy to defer to other opinions on the extension methods matter if you disagree on that.